nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -792,12 +792,14 @@ def _validate_and_override_config(self):
                 ) % vp_size == 0, 'Make sure the number of model chunks is the same across all pipeline stages.'
 
         if self.cfg.get('ub_tp_comm_overlap', False):
-            if not self.cfg.get('transformer_engine', False) or not self.cfg.get('sequence_parallel', False):
+            if self.cfg.get('ub_tp_comm_overlap', False) and not self.cfg.get('sequence_parallel', False):
```

Comment:
@erhoo82 `if self.cfg.get('ub_tp_comm_overlap', False)` seems to be redundant in L795 as it already exists in L794.